### PR TITLE
V2 postgrest sql

### PR DESCRIPTION
**Pull request by sd1li**
implements #97

### CODE REVIEW

File: models/postgrest/filteroperandtype_string.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Code generated by "stringer -type=FilterOperandType"; DO NOT EDIT.
```
Comment (poundifdef) [inline models/postgrest/filteroperandtype_string.go:1]: What is this for?

File: models/postgrest/sql.go [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+	sqlOpertors = map[string]sqlOpertor{
+		"eq":         {S: "="},
+		"gt":         {S: ">"},
+		"gte":        {S: ">="},
+		"lt":         {S: "<"},
+		"lte":        {S: "<="},
+		"neq":        {S: "<>"},
+		"like":       {S: "like"},
+		"ilike":      {S: "ilike"},
+		"match":      {S: "~"},
```
Comment (poundifdef) [inline models/postgrest/sql.go:34]: I think we can just keep eq, gt, gte, lt, lte, neq, like, ilike, in, is, not, or, and, all, any. For `all` and `any`, let's translate those into the equivalent AND/OR sql clauses.

File: models/postgrest/filteroperandtype_string.go (hunk 1 above)
Comment (sd1li) [reply models/postgrest/filteroperandtype_string.go:1]: Auto-geneated `FilterOperandType.String()` method, generated by `go generate`: https://github.com/scratchdata/ScratchDB/blob/d7ff953659eba70350ab6f8fb53f08a862167161/models/postgrest/postgrest.go#L1

File: models/postgrest/postgrest_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+			PopulateAST(testCase.Expression, root, parser.AST())
+
+			if testCase.Print {
+				parser.PrettyPrintSyntaxTree(testCase.Expression)
+
+				// j, err := json.MarshalIndent(root, "", "    ")
+				// if err != nil {
+				// 	t.Error(err)
+				// }
+				// log.Println(string(j))
```
Comment (chumaumenze) [inline models/postgrest/postgrest_test.go:78]: Is this useful?

File: cmd/api/postgrest.go [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+	"scratchdata/pkg/destinations"
+
+	"github.com/gofiber/fiber/v2"
+	"github.com/gofiber/fiber/v2/utils"
+	"github.com/rs/zerolog/log"
+)
+
+func (i *API) PostgrestQuery(c *fiber.Ctx) error {
+
+	apiKey := c.Locals("apiKey").(models.APIKey)
```
[the remark this answers is not in the document]
Comment (sd1li) [reply cmd/api/postgrest.go:17]: `apiKey` is set by `AuthMiddleware` so it should be fine here.

We could always use https://docs.gofiber.io/api/middleware/recover/ to recover from panics in general, but in this case I think a panic is desirable as it's a programmer error (the route is reachable without authentication).

**Review by chumaumenze — APPROVED** [review]

**Pull request closed without merging**